### DISCUSSION TITLE
npm run build fails when internet is not solid enough

### DISCUSSION THREAD

**Discussion by gramsco** [vercel/next.js#14328] [Help]
# Bug report

## Describe the bug

A colleague is facing an issue when building ("npm run build") the application on which we're both working.

The failure (see below) is produced by a mongodb client error (that we probably should handle anyway) and **I would like to know if internet connection can be a factor in the success and/or failure of the build process.**  

It is my main hypothesis, because ... 

- ... there is no deployment problem (vercel) and I can build the exact same code (including the required env variables to fetch the data) without any problem.

- ... yesterday she had a strong internet connection and everything went fine. Today, she has slow 3G at best. Slow but working: 'npm run dev' gets her all the data, she can push the code, etc. 

The error:
```
Automatically optimizing pages ..
Error occurred prerendering page "/forms/main/5ee6729eaab6a21787a25ef3". 
Read more: -error
TypeError: db.collection is not a function

 ```

## To Reproduce

I can't reproduce on my computer

## Expected behavior

npm run build should work the same with different internet connections

## System information

- OS: windows
- Version of Next.js: 9.4.0
- Version of Node.js: 12 LTS

## Additional context

1) The data is fetched from a heroku server

2) Geo _this one may not help ..._

Works at : 

Paris, France (good connection speed)
Latitude : 48.866667
Longitude : 2.333333

Does not work at : 

Ile de ré, France (an isle with pretty bad internet)
Latitude : 46.191111
Longitude :  -1.394463

**Comment by jamesmosier** [comment]
It sounds like your assumptions are correct. Assuming you are using something like `getStaticProps` and fetching from a database, if the database isn't available then a failure would make sense. 

Are you able to throw a catch/any error handling around the database connection that is having issues? That may help to narrow it down.

**Reply by gramsco** [reply]
Thanks for your reply. We'll try it asap and I'll edit with the catching!

**Reply by gramsco** [reply]
I edited the post. Even catching it's not really clear. I would expect a bad connection to slowly build the project and not to prevent it from building but maybe that's not possible, I don't know!

**Reply by jamesmosier** [reply]
I'm wondering if there is something within your Mongo connection that could do a retry if it fails due to slowness? I'm not sure what Next could do for this since it seems like the issue is an issue of database connection. 

Sorry i'm not much more help on that part!
